pkg/lock/lock.go

```diff
@@ -0,0 +1,22 @@
+package lock
+
+// ReadWrite returns a read/write lock for the given name
+func ReadWrite(domain string) ErrorRWLocker {
+	if c := getClient(); c != nil {
```

Comment:
As said before, if redis is not configured, calling `getClient` each time is costly for nothing.